internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/integration/utils/VeniceRouterWrapper.java

```diff
@@ -112,9 +116,12 @@ static StatefulServiceProvider<VeniceRouterWrapper> generateService(
     }
 
     return (serviceName, dataDirectory) -> {
-      String listenerPort =
-          properties.getOrDefault(ConfigKeys.LISTENER_PORT, Integer.toString(TestUtils.getFreePort())).toString();
-      int port = Integer.parseInt(listenerPort);
+      int port;
+      if (properties.containsKey(ROUTER_PORT_TO_USE_IN_VENICE_ROUTER_WRAPPER)) {
+        port = Integer.parseInt(properties.getProperty(ROUTER_PORT_TO_USE_IN_VENICE_ROUTER_WRAPPER));
+      } else {
+        port = TestUtils.getFreePort();
+      }
```

Comment:
Why does it help to define a new constant, rather than using `LISTENER_PORT`?

Reply:
`LISTENER_PORT` property is also used by VeniceServers. 

In VeniceServerWrapper we do explicitly add `LISTENER_PORT` property but this could be overridden by `LISTENER_PORT` from `configProperties`. `configProperties` is essentially `extraProperties` passed to all the components. 

So it's possible that both server and router will try to use this property

Reply:
I see, thanks! Maybe we should add a comment for this?

Reply:
Done!